config/initializers/rack_attack.rb

```diff
@@ -20,14 +23,23 @@ class Rack::Attack
   # Note: If you're serving assets through rack, those requests may be
   # counted by rack-attack and this throttle may be activated too
   # quickly. If so, enable the condition to exclude them from tracking.
+  #
+
+  # This stanza allows us to limit by which backend is selected by nginx.
+
+  ArchiveConfig.RATE_LIMIT_PER_NGINX_UPSTREAM.each do |k, _|
```

Comment:
If you use `do |k, v|` you can replace `ArchiveConfig.RATE_LIMIT_PER_NGINX_UPSTREAM[k]` with `v`.